providers/providers.go

```diff
@@ -152,7 +152,7 @@ func newProviderDataFromConfig(providerConfig options.Provider) (*ProviderData,
 		p.EmailClaim = providerConfig.OIDCConfig.UserIDClaim
 	}
 
-	if p.Scope == "" {
+	if providerConfig.Type == "oidc" && p.Scope == "" {
```

Comment:
Please consider to include also "keycloak-oidc" in this check otherwise this will generate an error using this provider since the scope in the login url will be empty and Keycloak is not happy with this behavior.
More could be useful to use ProviderType const for the check instead of the plain string.

Reply:
Hi @Kyserbyte 
I don't think that is a good idea. I would rather prefer to adapt the Keycloak OIDC provider to have a defaultScope. As this is the structure for all the other providers. I don't know why the defaultScope was not setup inside the provider itself.

Regarding the const, that is actual a good point. 

I'll raise a pull request addressing both of these issues.